### ISSUE TITLE
Indicate in test results if the emulator doesn't have the test-butler app installed

### ISSUE THREAD

**Issue by Kisty** [linkedin/test-butler#29]
Whenever I run the tests on an emulator without the test-butler app installed, it skips the test butler tests. There's no indication that they have been installed. Is it possible to insert in the exception message that the apk isn't installed.

```
:app:connectedDevDebugAndroidTest
Tests on Nexus_4_API_23_HARD(AVD) - 6.0 failed: Test run failed to complete. Expected 54 tests, received 49
:app:connectedDevDebugAndroidTest FAILED
```

**Comment by drewhannay** [comment]
Are you calling the TestButler methods in a `@Before` or `@BeforeClass` method? If I remember correctly, this is an issue with Google's test result parser where they don't properly display output for tests that fail before the `@Test` method starts. The exception you should be getting is [this one](https://github.com/linkedin/test-butler/blob/master/test-butler-library/src/main/java/com/linkedin/android/testbutler/TestButler.java#L237)

**Comment by Kisty** [comment]
Ah, right. Yes I am calling it in `@BeforeClass`. Let me see if I can repro it in the demo app.

**Comment by drewhannay** [comment]
Calling in `@Before` _**might**_ make the errors show up a bit more clearly, I don't remember for sure.

**Comment by Kisty** [comment]
Yes, I can confirm that putting the TestButler methods in `@Before` gives me that exception. 

In the [README](https://github.com/linkedin/test-butler/blob/master/README.md) it gives an example which does TestButler methods in `@BeforeClass`. Should we update the document to use `@Before` or just give a warning that `@BeforeClass` won't give an explanation that the APK isn't installed,

**Comment by drewhannay** [comment]
Yeah, changing the examples to `@Before` is probably a good idea. Want to send a pull request and I can merge it?

**Comment by Kisty** [comment]
Sure thing!

**Comment by Kisty** [comment]
Shall I also add the comment about not getting the exception message if doing stuff in `@BeforeClass` as another gotchya or do you think it's not needed.

**Comment by drewhannay** [comment]
Mmm, I'm fine either way. If you think it's helpful, sure.

**Comment by Kisty** [comment]
Related Android framework issue: https://code.google.com/p/android/issues/detail?id=183352

**Comment by drewhannay** [comment]
Interesting. Do you know what the exception looks like with `1.0.2` if you use `@BeforeClass`?
